internal/socket_unix.go

```diff
@@ -130,40 +130,67 @@ func connect(fd int, remoteAddr net.Addr, timeout time.Duration, opts ...sonicop
 		return err
 	}
 
-	if err := syscall.Connect(fd, ToSockaddr(remoteAddr)); err != nil {
-		// this can happen if the socket is nonblocking, so we fix it with a select
-		// https://man7.org/linux/man-pages/man2/connect.2.html#EINPROGRESS
+	for {
+		// Try to connect, if it succeeds we're done!
+		err := syscall.Connect(fd, ToSockaddr(remoteAddr))
+		if err == nil {
+			return nil
+		}
+
+		// Retry the connect syscall if interrupted
+		if err == syscall.EINTR {
```

Comment:
Another improvement is to test for `Errno`s correctly, according to the spec https://pkg.go.dev/syscall#Errno. This means using `errors.Is` pretty much everywhere in this function.